components/Carousel/Carousel.js

```diff
@@ -1,8 +1,41 @@
 class Carousel {
+    constructor(carouselElement){
+        this.carouselElement = carouselElement
+        this.imgs = carouselElement.querySelectorAll('img')
+        this.currentImg = 0
+        this.rightBtn = carouselElement.querySelector('.right-button')
+        this.leftBtn = carouselElement.querySelector('.left-button')
 
+        this.rightBtn.addEventListener('click', () => this.buttonClick(1))
+        this.leftBtn.addEventListener('click', () => this.buttonClick(-1))
+    }
+    /**
+     * @param {int} direction - either -1 or 1 for left and right respectively
+     */
+    buttonClick(direction){
+        
+        let prevImg = this.currentImg
+        let nextImg = this.currentImg + direction
+        if(nextImg < 0){
+            this.currentImg = this.imgs.length - 1
+        }else if( nextImg > this.imgs.length - 1){
+            this.currentImg = 0
+        }else{
+            this.currentImg = nextImg
+        }
+        console.log(prevImg,this.currentImg)
+        this.changeImg(prevImg, this.currentImg)
+    }
+
+    changeImg(prevImg, nextImg){
+        console.log(this.imgs[prevImg].classList)
+        this.imgs[prevImg].classList.toggle('active')
+        this.imgs[nextImg].classList.toggle('active')
```

Comment:
Great job!! Your carousel iterates backwards every fourth time. Why do you suppose this behavior is occurring?

Reply:
It's by design lol. It will show the first img if the next img 's index is more than the number of images. Since I'm not changing the order of the images, It will pop up to the left of the current img.